internal/lsp/server.go

```diff
@@ -115,9 +115,8 @@ type LanguageServer struct {
 
 // fileUpdateEvent is sent to a channel when an update is required for a file.
 type fileUpdateEvent struct {
-	Reason  string
-	URI     string
-	Content string
```

Comment:
This has been removed so it's no longer possible to send fileUpdateEvents with no content, this reduces the responsibility on the diagnostics worker. Now it can just update parses and send diagnostics.